spacy/coref_scorer.py

```diff
@@ -0,0 +1,124 @@
+# copied from coval
+# https://github.com/ns-moosavi/coval
+
+
+def get_cluster_info(predicted_clusters, gold_clusters):
+    p2g = get_markable_assignments(predicted_clusters, gold_clusters)
+    g2p = get_markable_assignments(gold_clusters, predicted_clusters)
+    # this is the data format used as input by the evaluator
+    return (gold_clusters, predicted_clusters, g2p, p2g)
+
+
+def get_markable_assignments(inp_clusters, out_clusters):
+    markable_cluster_ids = {}
+    out_dic = {}
+    for cluster_id, cluster in enumerate(out_clusters):
+        for m in cluster:
+            out_dic[m] = cluster_id
+
+    for cluster in inp_clusters:
+        for im in cluster:
+            for om in out_dic:
+                if im == om:
+                    markable_cluster_ids[im] = out_dic[om]
+                    break
+
+    return markable_cluster_ids
+
+
+def f1(p_num, p_den, r_num, r_den, beta=1):
+    p = 0 if p_den == 0 else p_num / float(p_den)
+    r = 0 if r_den == 0 else r_num / float(r_den)
+    return 0 if p + r == 0 else (1 + beta * beta) * p * r / (beta * beta * p + r)
+
```

Comment:
We don't typically use these `if`-`else` statements on one line throughout the core repo, so maybe rewrite them to separate `if` and `else` clauses?

(there are similar cases further down in this file too)

Reply:
Note that in general the style in this file doesn't look like ours because it's copy/pasted from another project, as indicated at the top of the file.

Reply:
Right, we definitely want to keep that note, but maybe have it read "adapted from" and then we're free to change this anyhow right, under their MIT license? It'll be part of the core repo so I do think its style should be changed slightly to be made coherent with our other code, as it's not unlikely that we'll want to edit this further at some point etc.

Reply:
Let's put the whole thing in `spacy/scorer.py` with license notes and the license added to `3rd_party_licenses.txt`? (Similar to the `roc_auc_score` that's already there.)

Reply:
Did that in this PR: https://github.com/explosion/spaCy/pull/10875